x-pack/plugins/security_solution/docs/prebuilt_rules_customization_rfc.md

```diff
@@ -0,0 +1,1301 @@
+    -   [Migration strategy: perform migration on all write operations of the rules](#migration-strategy-perform-migration-on-all-write-operations-of-the-rules){#toc-migration-strategy-perform-migration-on-all-write-operations-of-the-rules}
+    -   [Rule Management endpoints that should include this migration logic are:](#rule-management-endpoints-that-should-include-this-migration-logic-are){#toc-rule-management-endpoints-that-should-include-this-migration-logic-are}
+    -   [Other Rule Management endpoints that will need to be adapted to new schema, but will not need migration logic:](#other-rule-management-endpoints-that-will-need-to-be-adapted-to-new-schema-but-will-not-need-migration-logic){#toc-other-rule-management-endpoints-that-will-need-to-be-adapted-to-new-schema-but-will-not-need-migration-logic}
+    -   [Changes needed in `rule_monitoring` endpoints](#changes-needed-in-rule_monitoring-endpoints){#toc-changes-needed-in-rule_monitoring-endpoints}
+    -   [Technical implementation ofmigration](#technical-implementation-of-migration){#toc-technical-implementation-of-migration}
+-   [Exporting and importing
+    rules](#exporting-and-importing-rules){#toc-exporting-and-importing-rules}
+    -   [Exporting rules](#exporting-rules){#toc-exporting-rules}
+    -   [Importing rules](#importing-rules){#toc-importing-rules}
+-   [Customizing Prebuilt Rules](#customizing-prebuilt-rules){#toc-customizing-prebuilt-rules}
+    -   [Endpoints](#endpoints){#toc-endpoints}
+    -   [In the UI](#in-the-ui){#toc-in-the-ui}
+-   [List of things to fix (create tickets)](#list-of-things-to-fix-create-tickets){#toc-list-of-things-to-fix-create-tickets}
+    -   [Rule fields](#rule-fields){#toc-rule-fields}
+-   [Updating Prebuilt Rules](#updating-prebuilt-rules){#toc-updating-prebuilt-rules}
+    -   [Changes to upgrade `_review` and `_perform` endpoints](#changes-to-upgrade-_review-and-_perform-endpoints){#toc-changes-to-upgrade-_review-and-_perform-endpoints}
+    -   [Rule field diff algorithms](#rule-field-diff-algorithms){#toc-rule-field-diff-algorithms}
+-   [Changes in UI](#changes-in-ui){#toc-changes-in-ui}
+-   [Scenarios for bulk accepting updates](#scenarios-for-bulk-accepting-updates){#toc-scenarios-for-bulk-accepting-updates}
+-   [Marking rules (and their fields) as customized](#marking-rules-and-their-fields-as-customized){#toc-marking-rules-and-their-fields-as-customized}
+-   [Other open questions](#other-open-questions){#toc-other-open-questions}
+
+
+
+## Necessary rule schema changes
+
+In order to support the customization of Elastic Prebuilt Rules, we need to modify our rule schema. This involves introducing two new fields: `prebuilt` and `customized`. Simultaneously, we need to deprecate the existing `immutable` field, which determines if a rule is an Elastic Prebuilt Rule and whether its fields can be modified/customized.
```

Comment:
Yes. All our rules will be "mutable" after this epic, so the field name `immutable` will lose its meaning over time and become confusing, especially for consumers of the API. That's why we want to eventually deprecate it and fully replace it with `prebuilt`.

I think the change is necessary and worth the effort, but since you are right that is only a "name" change, I'm interested in your opinion as well.

Meanwhile, I'll add this explanation to the RFC in this section.

Reply:
Actually, the `prebuilt` field should replace the `immutable` field but with one difference: the `prebuilt` field will mark if the rule is an Elastic Prebuilt Rule or not, but now all rules will be able to be customized by the user, i.e. independently of the value of `prebuilt`. That's also a semantic difference between the two fields.